### PR TITLE
restore onNewViewRequested

### PR DESCRIPTION
**Pull request by howard-stearns**
The ability to handle Web page requests for new tabs was lost in WebView.qml, I think is part of the marketplace development.

### CODE REVIEW

**Comment by howard-stearns** [comment]
I've confirmed that this restores the ability to handle web pages that request a new url.
However, please test to make sure that restoring does not break any of the intended new marketplace functionality.

**Comment by howard-stearns** [comment]
@elisa-lj11, can you check this for me?  (I tried to assign to you, but github isn't letting me do so.)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6272db42dafc2cbc0171f81d21eb80bffb817584 available at: https://gist.github.com/ca5d75d6593ca65d8f9b1fab4adbdef7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-6272db4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-6272db4.dmg

**Comment by howard-stearns** [comment]
test this please

**Comment by elisa-lj11** [comment]
Clara.io has retained almost all of its functionality: the only part that isn't working is the automatic new browser closure. I am looking into this now.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6272db42dafc2cbc0171f81d21eb80bffb817584 available at: https://gist.github.com/ca5d75d6593ca65d8f9b1fab4adbdef7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-6272db4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-6272db4.dmg

**Comment by howard-stearns** [comment]
I've added a hook that gets called with the new component, window, and request, and does nothing with them. @elisa-lj11, you can override this hook in your client to capture the window in a timer callback.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 11f00ed46cf2d571cf6bc2f3e72f0efb107a8069 available at: https://gist.github.com/ca5d75d6593ca65d8f9b1fab4adbdef7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-11f00ed.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-11f00ed.dmg

**Comment by howard-stearns** [comment]
I'm holding this until it incorporates the marketplace side of this.

https://highfidelity.fogbugz.com/f/cases/1594/WebView-qml-no-longer-handles-new-tab-requests

**Comment by howard-stearns** [comment]
Now includes the marketplace changes necessary to match

**Comment by hifi-gustavo** [comment]
Build console output for commit: c5b998a2a8e7472ebcbf5dffe3a272308be5eb9a available at: https://gist.github.com/ca5d75d6593ca65d8f9b1fab4adbdef7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-c5b998a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-c5b998a.dmg

**Comment by elisa-lj11** [comment]
I just tested downloading a shield model and triggering a new webview from a regular browser, and both work as intended.

**Comment by howard-stearns** [comment]
I've confirmed that the WebView part works, so between that and Elisa's comments, I'll call this a pass.

File: interface/resources/qml/controls-uit/WebView.qml [hunk 1]
```diff
@@ -61,6 +61,13 @@ WebEngineView {
         }
     }
 
+    property var newWindowHook: function (component, newWindow, request) { }; // override if you need to
+    onNewViewRequested:{
+        var component = Qt.createComponent("../Browser.qml");
+        var newWindow = component.createObject(desktop);
+        request.openIn(newWindow.webView)
+        newWindowHook(component, newWindow, request);
+    }
```
Comment (howard-stearns) [inline interface/resources/qml/controls-uit/WebView.qml]: This handler is what got inadvertently removed in the markeplace work -- except for the call to newWindowHook, which is added now to make the markeplace stuff work.

File: interface/resources/qml/MarketplaceComboBox.qml [hunk 2]
```diff
@@ -48,15 +48,12 @@ Rectangle {
...
-            request.openIn(newWindow.webView);
+        newWindowHook: function (component, newWindow) {
             if (File.isZippedFbx(desktop.currentUrl)) {
+                runJavaScript(autoCancel);
                 zipTimer.handler = function() {
                     newWindow.destroy();
-                    runJavaScript(autoCancel);
-                }
+                };
                 zipTimer.start();
```
Comment (howard-stearns) [inline interface/resources/qml/MarketplaceComboBox.qml]: The stuff up through request.openIn belongs in WebView (and now is there with this pr).
The rest of the code is as before -- except for the ordering of runJavaScript, which is a change per Elisa.

**Comment by howard-stearns** [comment]
Following discussion with @jherico, we've decided to do this a different way, which requires retesting.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88245075233464e1d860e977a428652ea0aea53f available at: https://gist.github.com/ca5d75d6593ca65d8f9b1fab4adbdef7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-8824507.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8540-8824507.dmg

**Comment by elisa-lj11** [comment]
Clara is still functioning as intended.

**Comment by howard-stearns** [comment]
as is the the ordinary new windows initiated from WebView. Re-passes.

**Pull request merged**